ncm-iptables/src/main/perl/iptables.pm

```diff
@@ -636,101 +611,84 @@ sub rule_options_translate {
-	$? = 1;
-	$@ = 'filename to write missing';
-	return $?;
+    if (!$filename) {
+        $? = 1;
+        $@ = 'filename to write missing';
+        return $?;
     }
 
     # Open the file.
-    unless ( open(FILE, ">$filename") ) {
-	$? = 6;
-	$@ = "cannot open $filename";
-	return $?;
-    }
+    my $fh = CAF::FileWriter->open(
+        $filename,
+        owner => 'root',
+        group => 'root',
+        mode => '0444',
+    );
+
     # write our "tag" into it. Assist some poor admin in debugging..
-    print FILE "# Firewall configuration written by ncm-iptables\n";
-    print FILE "#  Manual modifications will be overwritten on the next NCM run.\n";
+    print $fh "# Firewall configuration written by ncm-iptables\n";
+    print $fh "# Manual modifications will be overwritten on the next NCM run.\n";
+    $self->debug(5, "Wrote header tag to file");
 
     # Write new content to file.
-    if ( defined $iptables && ref($iptables) =~ /^HASH/ ) {
-	foreach $table (keys %iptables_totality) {
-	    next if ( ! defined $iptables->{$table} || $iptables->{$table} eq "" ||
-		      ref($iptables->{$table}) !~ /^HASH/ );
-
-	    print FILE "*$table\n";
-
-
-	    if ( defined $iptables->{$table}->{preamble} &&
-		 ref($iptables->{$table}->{preamble}) =~ /^HASH/ ) {
-		my $preamble = $iptables->{$table}->{preamble};
-
-
-
-		foreach $chain ( @{$iptables_totality{$table}{chains}}) {
-		    next if ( ! defined $preamble->{$chain} || $preamble->{$chain} eq "" );
-		    my $g = $chain;
-		    $g =~ tr/a-z/A-Z/;
-		    $preamble->{$chain} =~ s/^[\s\t]*|[\s\t]*$//g;
-		    $preamble->{$chain} =~ s/[\s\t+]/ /g;
-		    print FILE ":$g $preamble->{$chain}\n";
-		}
-	    }
-
-
-
-	    foreach $target ( sort keys %{$iptables_totality{$table}{user_targets}} ){
-		print FILE "-N $target\n";
-	    }
-
-	    foreach $target (@{$iptables_totality{$table}{targets}}) {
-
-		next if ( ! defined $iptables->{$table}->{rules}->{$target}         );
-		next if (   ref($iptables->{$table}->{rules}->{$target}) !~ /^HASH/ );
-		next if ( ! scalar(%{$iptables->{$table}->{rules}->{$target}})      );
-
-		foreach $name (sort { $a <=> $b; } keys %{$iptables->{$table}->{rules}->{$target}}) {
-
-		    next if ( $name !~ /^\d+$/ );
-
-		    $rule = $iptables->{$table}->{rules}->{$target}->{$name};
-		    $line = '';
-		    foreach $field (&sort_keys($self,$rule)) {
-			$line .= ($line) ? " $field" : $field;
-			$line .= " $rule->{$field}" if $options_arg{$field};
-		    }
-		    print FILE "$line\n" if $line and $line !~ /^-N/;
-
-		}
-
-	    }
-
-	    print FILE "$iptables->{$table}->{epilogue}\n"
-		if ( defined $iptables->{$table}->{epilogue} && $iptables->{$table}->{epilogue} ne "");
-
-	}
-    }
-
-    # Close the temporary file.
-    unless ( close(FILE) ) {
-	$? = 7;
-	$@ = "cannot close $filename";
-	return $?;
-    }
-
-    $? = 0;
-    $@ = "modified $filename";
-    return $?;
+    if (defined $iptables && ref($iptables) =~ /^HASH/) {
```

Comment:
i would do a 
```perl
if(! $iptables) {
    $self->info("No iptables passed");
    return 0;
}
```
before opening initialising the `FileWriter` to avoid the indentation
(better yet, do the cehck before the method is called)